pkg/cmd/util/clientcmd/factory.go

```diff
@@ -367,57 +351,54 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	}
 	kLogsForObjectFunc := w.Factory.LogsForObject
 	w.LogsForObject = func(object, options runtime.Object) (*restclient.Request, error) {
-		switch t := object.(type) {
-		case *deployapi.DeploymentConfig:
-			dopts, ok := options.(*deployapi.DeploymentLogOptions)
-			if !ok {
-				return nil, errors.New("provided options object is not a DeploymentLogOptions")
-			}
-			oc, _, err := w.Clients()
-			if err != nil {
-				return nil, err
-			}
-			return oc.DeploymentLogs(t.Namespace).Get(t.Name, *dopts), nil
-		case *buildapi.Build:
-			bopts, ok := options.(*buildapi.BuildLogOptions)
-			if !ok {
-				return nil, errors.New("provided options object is not a BuildLogOptions")
-			}
-			if bopts.Version != nil {
-				return nil, errors.New("cannot specify a version and a build")
-			}
-			oc, _, err := w.Clients()
-			if err != nil {
-				return nil, err
-			}
-			return oc.BuildLogs(t.Namespace).Get(t.Name, *bopts), nil
-		case *buildapi.BuildConfig:
-			bopts, ok := options.(*buildapi.BuildLogOptions)
-			if !ok {
-				return nil, errors.New("provided options object is not a BuildLogOptions")
-			}
+		gvk, _, err := api.Scheme.ObjectKind(object)
+		if err != nil {
+			return nil, err
+		}
+		if latest.OriginKind(gvk) {
```

Comment:
Hm, it seems you already have a gvk but it never enters this scope for deployment logs (and I guess for build logs as well hence the conformance test failures)